### PR TITLE
[PM-11698] - fix header in billing sync dialog

### PR DESCRIPTION
**Pull request by jaasen-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11698

## 📔 Objective

This PR fixes the header size in the billing sync dialogs.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: apps/web/src/app/billing/organizations/billing-sync-api-key.component.html [hunk 1]
```diff
@@ -1,8 +1,8 @@
 <form [formGroup]="formGroup" [bitSubmit]="submit">
   <bit-dialog>
-    <h1 bitDialogTitle>
+    <span bitDialogTitle>
```
Comment (danielleflinn) [inline apps/web/src/app/billing/organizations/billing-sync-api-key.component.html:3]: I don't think changing this to a `<span>` is the proper solution as this is a new view's title it should remain an `<h1>` so a screen reader can navigate the new view by heading.

Looking at Storybook it looks like we are missing some of the `bittypography="h3"` styling that is usually applied to the `bitDialogTitle` 

@vleague2 would you mind weighing in on why the Component Library is not applying the proper heading styles here?

File: apps/web/src/app/billing/organizations/billing-sync-key.component.html [hunk 2]
```diff
@@ -1,8 +1,8 @@
 <form [formGroup]="formGroup" [bitSubmit]="submit">
   <bit-dialog>
-    <h1 bitDialogTitle>
+    <span bitDialogTitle>
```
Comment (danielleflinn) [inline apps/web/src/app/billing/organizations/billing-sync-key.component.html:3]: same as previous comment

**Review by danielleflinn — CHANGES_REQUESTED** [review]

File: apps/web/src/app/billing/organizations/billing-sync-api-key.component.html (hunk 1 above)
Comment (danielleflinn) [reply apps/web/src/app/billing/organizations/billing-sync-api-key.component.html:3]: @jaasen-livefront Which dialogs were you referencing?  The ones I found had `<h1><span></span></h1> `

Comment (vleague2) [reply apps/web/src/app/billing/organizations/billing-sync-api-key.component.html:3]: @danielleflinn @jaasen-livefront It looks like the `bitDialogTitle` is just a selector used by the dialog component that doesn't apply any styles. The `bitDialogTitle` gets wrapped by an `h1` applying `h3` styles inside the dialog component itself, so I think this change is correct. What story are you looking at where the styling is missing?

Comment (danielleflinn) [reply apps/web/src/app/billing/organizations/billing-sync-api-key.component.html:3]: Ok so the rendered dialog will include the h1 in the browser inspector? That's all I care about but its hard to tell if that is happening here

**Review by danielleflinn — APPROVED** [review]
Ignore previous comments if this renders with an `<h1>` element and proper styling

**Comment by jaasen-livefront** [comment]
> Ignore previous comments if this renders with an `<h1>` element and proper styling

@danielleflinn Indeed! The `span` is wrapped within an `h1` tag.

**Review by vleague2 — APPROVED** [review]

**Review by jonashendrickx — APPROVED** [review]

**Pull request merged**